src/toHaveStyleRule.js

```diff
@@ -54,28 +54,49 @@ const getAtRules = (ast, options) => {
     .reduce((acc, rules) => acc.concat(rules), [])
 }
 
-const getModifiedClassName = (className, modifier = '') => {
-  const classNameSelector = `.${className}`
-  let prefix = ''
-
-  modifier = modifier.trim()
-  if (modifier.includes('&')) {
-    modifier = modifier.replace(/&/g, classNameSelector)
-  } else {
-    prefix += classNameSelector
-  }
-  const first = modifier[0]
-  if (first !== ':' && first !== '[') {
-    prefix += ' '
-  }
-
-  return `${prefix}${modifier}`.trim()
+const getModifiedClassName = ({ className, modifier = '', componentId }) => {
```

Comment:
I would revert this and remove the `& &` test case.

Previously, `& &` was rendered as:

```css
.lnlmMg .lnlmMg {
  color: deepskyblue;
}
```

While now it's rendered as:

```css
.icfrky .sc-jTzLTM {
  color: deepskyblue;
}
```

I wonder if this is intentional and what are the motivations behind this decision.
We should open an issue on the main repo.